docs/releasing.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Releasing
+weight: 20
+description: >
+  How to release Inspektor Gadget
+---
+
+The Inspektor Gadget release process is heavily automated, however some manual steps are still
+required. This guide describes the process to make a new release of Inspektor Gadget and includes
+some information of what to do in case something goes wrong.
+
+Be sure you have a general understading of the whole release process before starting to follow the
+steps described here. It's also important to stay aware as a mistake could cause issues for our
+users.
+
+## Release notes
+
+To get a draft of the release notes we use the [generate release](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes) notes from Github:
+
+1. Go to https://github.com/inspektor-gadget/inspektor-gadget/releases
+2. Click `Draft a new release`
+3. Create a new tag. Note that it won't be actually created.
+![Create Release Tag](images/release-create-tag.png)
+4. Click `Generate release notes`
+5. **DON'T publish the release nor save draft**, this will be done automatically in next steps.
+Copy and paste the generated release notes to a text editor and update them to match the format used
+for other releases:
+
+```
+< Relevant changes >
+
+### General Improvements
+### Bug Fixes
+### Documentation Improvements
+### Testing and Continue Integration
+```
+
+The release notes will be used in the following steps.
+
+## Making a new release
+
+0. Be sure tests on main branch are passing, if they are not passing it's needed to fix them before.
+1. Switch to the main branch and pull latest changes
+
+```bash
+$ git checkout main
+$ git pull origin
+```
+
+Double check that the latest commit is the right one to make the release.
+
+2. Tag a release canditate version
+
+```bash
+$ git tag v0.x.0-rc
+```
+
+3. Push tag to remote
+
+```bash
+$ git push origin v0.x.0-rc
```

Comment:
@eiffel-fl I noticed it would create a release on Github. I had to remove this by hand later on. Does anybody have an idea how to fix it?

Reply:
You could add the following in the relevant Github action. We already do something similar for the krew-index action.
```diff
-if: startsWith(github.ref, 'refs/tags/v')
+if: startsWith(github.ref, 'refs/tags/v') && !endsWith(github.ref_name, 'rc')
```

Reply:
That can work, but then what's the point of creating a release candidate?

IMO we can avoid this step and test changes to the release job in a fork. This to avoid notifying users about a rc candidate version. @eiffel-fl what do you think?

Reply:
> The whole point for me is to test and ensure everything is OK.

Almost all the testing is done in main when a PR is merged. If the tests are passing in main, the only thing that could break is the release job. In that case I'd suggest to test changes modifying it in a fork before merging them.  

> Will you be able to test on AKS by using a fork? 

That's right, one should configure the secrets in the repo to do it.

> I do not think this a bad thing to have release candidate available for everyone.

I agree that release candidates are good, however I think they bring little value as we're using them in our project: to check if the release pipeline is working fine.

A release candidate is a version that you release, let some users test it and then you fix the bugs they find, however in our case we're just release the final version after the rc. 


In summary, I feel creating a release candidate adds extra complication and noise to the release and doesn't bring very big benefits, hence I think we should remove this. What do you think?